### PR TITLE
Add gcsio version as usre agent for gRPC

### PR DESCRIPTION
**Pull request by hongyegong**

### CODE REVIEW

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/StorageStubProvider.java [hunk 1]
```diff
@@ -53,6 +54,13 @@
   private final ExecutorService backgroundTasksThreadPool;
   private final List<ChannelAndRequestCounter> mediaChannelPool;
 
+  private static final String PROPERTIES_FILE = "gcsio.properties";
+  private static final String VERSION_PROPERTY = "gcsio.version";
+  private static final String UNKNOWN_VERSION = "0.0.0";
+
+  private static final String VERSION = PropertyUtil.getPropertyOrDefault(
+      StorageStubProvider.class, PROPERTIES_FILE, VERSION_PROPERTY, UNKNOWN_VERSION);
+
```
Comment (medb) [inline gcsio/src/main/java/com/google/cloud/hadoop/gcsio/StorageStubProvider.java:63]: Can we reuse the same user agent as we use for HTTP requests?

**Review by medb — CHANGES_REQUESTED** [review]

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/StorageStubProvider.java (hunk 1 above)
Comment (hongyegong) [reply gcsio/src/main/java/com/google/cloud/hadoop/gcsio/StorageStubProvider.java:63]: In theory we could, but that will need to add irrelevant properties in GCSReadOptions in order to plumb that piece of data thru. This is just following the same way we do in other modules.

ALso depends on what we want to include here, userAgent for http will include other infomation I think which might not apply here.

Comment (medb) [reply gcsio/src/main/java/com/google/cloud/hadoop/gcsio/StorageStubProvider.java:63]: I think that we need to include the same user agent in gRPC requests for consistency with GCS requests.

Probably issue here is that we initialize stub in the gRPC channels whereas HTTP channels accept pre-configured `storage` object?

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/StorageStubProvider.java [hunk 2]
```diff
@@ -53,6 +53,8 @@
   private final ExecutorService backgroundTasksThreadPool;
   private final List<ChannelAndRequestCounter> mediaChannelPool;
 
+  private String userAgent;
```
Comment (medb) [inline gcsio/src/main/java/com/google/cloud/hadoop/gcsio/StorageStubProvider.java:56]: Make it `final`

**Review by medb — APPROVED** [review]

**Pull request merged**